### DISCUSSION TITLE
train_rave.py - Unrecognized Argument Error

### DISCUSSION THREAD

**Discussion by ianberman** [acids-ircam/RAVE#34] [Q&A]
Hi! Super interested in this project as I've previously played around with realtime timbre transfer using mubu knn's in max. I'm rather new to cli/python stuff though, so apologies if this is a basic question!

I'm getting an error in `train_rave.py` as follows: `error: unrecognized arguments: /venv_test_train/rave`

I called my `--name`  "venv_test_train" because I set up a pip venv to do everything after encountering a similar error when trying to install everything from `requirements.txt` globally. I am just using the cli-helper script and leaving all the parameters as default. 

I created the venv with python 3.10, otherwise everything is clean. Any chance I just need to use 3.9? Otherwise I am rather lost. (also I am on OSX Intel 10.15)

Thanks, any clues appreciated :)

<img width="621" alt="Screen Shot 2022-04-06 at 3 08 28 PM" src="https://user-images.githubusercontent.com/32076251/161983258-95a36329-71c8-44c5-93fc-b573999b680f.png">

**Comment by branlsnyder** [comment]
Hi ianberman,

I did not have this exact error. But my issue with installing from requirements.txt was solved when I used python 3.9 instead of 3.10. 

I also needed to install an older version of torchmetrics (0.6.2), but that might be unrelated to your issue.

**Reply by vigliensoni** [reply]
Get rid of the space after `TEMP_FOLDERS` so that your ```--preprocessed``` path is 

`/Users/..../TEMP_FOLDERS/test_train_3/rave`

**Reply by ianberman** [reply]
Ah, I just came to the same conclusion and fixed it, thank you!! Defaulted to the space when I dropped the folder in via finder.

Unfortunately now I am getting the `TypeError: cannot pickle 'Environment' object` issue, since I'm on Mac. I didn't realize that would present an issue until now, but it seems documentation has been updated.

**Comment by ianberman** [comment]
I'm curious if anyone more knowledgeable than me knows if it's possible, or economically reasonable, to use cloud solutions such as Google Colab to train models?

edit: Colab and Kaggle use python 3.7 and it seems like a hassle to try to change that. looking into alternatives before I buy a Nvidia card, but believe me I don't need much of an excuse :) 
via colab:
`[libprotobuf FATAL google/protobuf/stubs/common.cc:87] This program was compiled against version 3.9.2 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.17.3).  Contact the program author for an update.  If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library.  (Version verification failed in "bazel-out/k8-opt/bin/tensorflow/core/framework/tensor_shape.pb.cc".)
terminate called after throwing an instance of 'google::protobuf::FatalException'`

Curious if RAVE is dependent on functionality of 3.9 or if there's a way to use 3.7? Totally out of my depth here, just wondering. I'm sure it would make many Mac users happy.
